server/bin/pbench-tarball-stats

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""Pbench Tar Ball Stats
+
+Scan through the ARCHIVE hierarchy for tar balls and report some
+statistics on them:
+
+  * Total number of controllers
+  * Tar balls per controller (most to least)
+  * Tar balls by month for all years, count
+  * Tar balls by month for all years, total size
+  * Top 10 most recent controllers
+  * Top 10 largest tar balls
+"""
+
+import collections
+import locale
+import os
+import re
+import sys
+
+from datetime import datetime
+from argparse import ArgumentParser
+
+import pbench
+from pbench import (
+    PbenchConfig,
+    BadConfig,
+    get_pbench_logger,
+)
+
+
+_NAME_ = "pbench-tarball-stats"
+
+tb_pat_r = (
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+tb_pat = re.compile(tb_pat_r)
+
+
+def gen_tar_balls(archive):
```

Comment:
This scanner is almost the same as the audit generator scan, and the indexer scan, and the importer scan I hacked up a while ago ... one of these days we really need to factor out an extensible minimum base class that we can just get right once.

Or maybe we'll just keep doing this until we switch to object store and get rid of the crufty directory hierarchy "metadata" model entirely ...

Reply:
Yes, I've moved this PR to the `b0.69` branch, since we can use the data to report all this data in the `main` branch.

So for now, we'll just keep this scanner.